rust/toolchain.bzl

```diff
@@ -126,6 +132,23 @@ def _make_libstd_and_allocator_ccinfo(ctx, rust_lib, allocator_library):
         """).format(ctx.label, ctx.attr.rust_lib))
     rust_stdlib_info = ctx.attr.rust_lib[rust_common.stdlib_info]
 
+    if rust_stdlib_info.self_contained_files:
+        compilation_outputs = cc_common.create_compilation_outputs(
+            objects = depset(rust_stdlib_info.self_contained_files),
+        )
+
+        linking_context, _linking_outputs = cc_common.create_linking_context_from_compilation_outputs(
```

Comment:
So this will register an archiving action that will put all objects into the archive. Is it intended? Alternatively you can put those objects into https://docs.bazel.build/versions/3.5.0/skylark/lib/LibraryToLink.html#objects. I'm not sure if ld.exe will be able to use `--start-lib/--end-lib` for those objects though. Gold and lld will be fine with not having a static library.

Reply:
I think you'd know better than I here. The goal was to find a mechanism that wires up these object files together with the existing `CcInfo` provider that was being generated using [cc_common](https://docs.bazel.build/versions/4.0.0/skylark/lib/cc_common.html). Is there a more direct path to achieve this?

Reply:
I looked around and it seems what you do is correct :)